MetaMask/metamask-extension
# Add simplified version of Snap install warning modal

## **Description**
This PR updates the popup/modal shown on Snap install when installing a Snap with more powerful permissions. UI/UX changes are done according to the designs proposed on Figma: https://www.figma.com/file/Wlwk0cTx6PQkNhiX7vN13e/Snaps-Platform-v1?node-id=0%3A1&mode=dev

The following changes are done:
* Removed warning `message` constructed for key entropy permissions inside `ui/helpers/utils/permission.js` and replaced with simplifed version under `warningMessageSubject` property.
* Replaced deprecated UI components.
* Changed icon used for tooltip on more powerful permissions from triangle warning `/!\` to simple info icon `(i)`. Colors of the icons in specific use cases are left unchanged. (Requested by @eriknson).
* Updated permission name representation in the UI for `BIP32` and `BIP44` (Requested by @eriknson).
* Updated permission name representation in the UI for `BIP32` and `BIP44` for unknown networks (coin type) (Requested by @FrederikBolding and @eriknson on Slack).
* Updated description for both `BIP32` and `BIP44` permissions (using consolidated description for both, other are redundant and removed) (Requested by @eriknson).
* Other minor code refactorings.
* Remove outdated translations.

## **Manual testing steps**
1. Go to Snaps Directory and try installing a Snap with key entropy permissions (BIP-44 or BIP-32) (i.e. https://snaps.metamask.io/snap/npm/cosmsnap/snap/).
2. When on the permission page in the MetaMask extension popup, click `Install` and warning popup should open. Review the content and UI elements.
3. Click the new checkbox to agree with the statement above.
4. Click `Confirm`.
5. Installation should continue.
6. Snap should be successfully installed.

## **Screenshots/Recordings**
### **Before**
![Screenshot 2023-10-24 at 17 09 45](https://github.com/MetaMask/metamask-extension/assets/13301024/9db06890-2123-4722-8958-772621ae23d2)
![Screenshot 2023-10-24 at 17 10 37](https://github.com/MetaMask/metamask-extension/assets/13301024/750c4b05-caae-4b28-a509-df17046933d9)
![Screenshot 2023-10-25 at 11 06 10](https://github.com/MetaMask/metamask-extension/assets/13301024/042c7e38-0f91-4fdd-aaa5-66ef31bb3b95)

### **After**
![Screenshot 2023-11-01 at 12 19 54](https://github.com/MetaMask/metamask-extension/assets/13301024/844366eb-7195-45e5-ba39-7fd79bb8aae7)
![Screenshot 2023-11-01 at 12 21 44](https://github.com/MetaMask/metamask-extension/assets/13301024/25334556-33e5-4c52-a6fa-29cbf5ede386)
![Screenshot 2023-11-01 at 12 20 21](https://github.com/MetaMask/metamask-extension/assets/13301024/38394c06-145f-4162-afbf-a3d773ab9ba8)
![Screenshot 2023-11-01 at 12 22 12](https://github.com/MetaMask/metamask-extension/assets/13301024/885b80cf-4381-4f85-8efc-401d39b79b45)
![Screenshot 2023-11-01 at 12 23 54](https://github.com/MetaMask/metamask-extension/assets/13301024/1c877aaa-7400-40fd-989f-280d34b9881a)
![Screenshot 2023-11-01 at 12 23 38](https://github.com/MetaMask/metamask-extension/assets/13301024/5a863ae9-b9f4-4dc0-93d3-885e0c6ca359)
![Screenshot 2023-11-01 at 12 24 18](https://github.com/MetaMask/metamask-extension/assets/13301024/b2d3de80-6119-4be0-b8a7-38026e7efd11)
![Screenshot 2023-11-01 at 12 20 40](https://github.com/MetaMask/metamask-extension/assets/13301024/741aad1a-a29a-4457-82bf-468bba01ebcd)
![Screenshot 2023-11-01 at 12 23 15](https://github.com/MetaMask/metamask-extension/assets/13301024/c80f1300-35d4-45c3-a31c-56abca0c0f01)
![Screenshot 2023-11-01 at 12 30 52](https://github.com/MetaMask/metamask-extension/assets/13301024/b40a9db0-b2d3-448c-b495-cbb674132714)
![Screenshot 2023-11-02 at 17 49 37](https://github.com/MetaMask/metamask-extension/assets/13301024/5ddd9986-0680-4138-89fc-78d4cc06d570)
![Screenshot 2023-11-02 at 17 50 11](https://github.com/MetaMask/metamask-extension/assets/13301024/765be3eb-1605-4c9e-b564-c24eca0d2e43)
![Screenshot 2023-11-02 at 17 51 20](https://github.com/MetaMask/metamask-extension/assets/13301024/c0a26733-d86b-4f9f-b7f4-e0b60999cbda)

## **Related issues**
Fixes: https://github.com/MetaMask/snaps/issues/1807

## **Pre-merge author checklist**

## **Pre-merge reviewer checklist**
